src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
Yes, my statement was imprecise (pun intended). I meant the "maximum precision available by a 64-bit floating point number". If the mantissa is 52 bits, that's still enough precision for to support uv__hrtime. 
In conclusion, if you want to implement uv__hrtime using 128-bit integer math, you can certainly do that. I just don't think that it's going to fix the issue of timers firing early (https://github.com/joyent/node/issues/8960).